apps/worker/tasks/base.py

```diff
@@ -219,77 +229,99 @@ def apply_async(self, args=None, kwargs=None, **options):
         }
         return super().apply_async(args=args, kwargs=kwargs, headers=headers, **options)
 
-    # Called when attempting to retry the task on db error
-    def _retry(self, countdown=None):
-        if not countdown:
-            countdown = self.default_retry_delay
-
-        try:
-            self.retry(countdown=countdown)
-        except MaxRetriesExceededError:
-            if UploadFlow.has_begun():
-                UploadFlow.log(UploadFlow.UNCAUGHT_RETRY_EXCEPTION)
-            if TestResultsFlow.has_begun():
-                TestResultsFlow.log(TestResultsFlow.UNCAUGHT_RETRY_EXCEPTION)
-
-    def safe_retry(self, max_retries=None, countdown=None, exc=None, **kwargs):
+    def retry(
+        self,
+        countdown=None,
+        exc=None,
+        kwargs=None,
+        max_retries=_NOT_PROVIDED,
+        **other_kwargs,
+    ):
         """
-        Safely retry with max retry limit and proper metrics tracking.
-
-        This method provides a common retry pattern for all tasks with:
-        - Configurable max retry limit
-        - Automatic metric tracking
-        - Consistent error handling
+        Override Celery's retry() to automatically check max_retries and track metrics.
+        If max_retries is not provided, uses self.max_retries from the task class.
+        If max_retries=None is explicitly passed, means unlimited retries (matches Celery behavior).
 
         Args:
-            max_retries: Maximum number of retries allowed (default: task's max_retries)
-            countdown: Seconds to wait before retry (default: exponential backoff)
+            countdown: Seconds to wait before retry (optional)
             exc: Exception to include in retry (optional)
-            **kwargs: Additional kwargs to pass to self.retry()
+            kwargs: Task kwargs to use when retrying (optional, Celery-specific parameter)
+            max_retries: Maximum number of retries allowed (default: task's max_retries).
+                        Pass None explicitly for unlimited retries (matches Celery behavior).
+            **other_kwargs: Additional kwargs to pass to Celery's retry()
+        """
+        request = getattr(self, "request", None)
+        if request is None:
+            # If no request, can't retry - this shouldn't happen in normal operation
+            # but handle gracefully by using default values
+            current_retries = 0
+            request_kwargs = {}
+        else:
+            current_retries = request.retries if hasattr(request, "retries") else 0
+            request_kwargs = (
+                request.kwargs if hasattr(request, "kwargs") and request.kwargs else {}
+            )
 
-        Returns:
-            True if retry was scheduled
-            False if max retries exceeded
+        if max_retries is _NOT_PROVIDED:
+            task_max_retries = self.max_retries
+        else:
+            task_max_retries = max_retries
```

Comment:
maybe just `task_max_retries = self.max_retries if max_retries is NOT_PROVIDED else max_retries` or similar